bioblend/galaxy/container_resolution/__init__.py

```diff
@@ -0,0 +1,238 @@
+            'requirements': [{'name': 'pyarrow', 'specs': [], 'type': 'package', 'version': '4.0.1'}],
+            'status': {
+                'cacheable': False,
+                'container_description': {'identifier': 'quay.io/biocontainers/pyarrow:4.0.1', 'resolve_dependencies': False, 'shell': '/bin/bash', 'type': 'docker'},
+                'container_resolver': {'builds_on_resolution': False, 'can_uninstall_dependencies': False, 'model_class': 'MulledDockerContainerResolver', 'resolver_type': 'mulled'},
+                'dependency_type': 'docker',
+                ...
+            },
+            'tool_id': 'CONVERTER_parquet_to_csv'
+        }
+        """
+        params = {}
+        if tool_id:
+            params["tool_id"] = tool_id
+        if resolver_type:
+            params["resolver_type"] = resolver_type
+        if container_type:
+            params["container_type"] = container_type
+        params["requirements_only"] = str(requirements_only)
+        params["install"] = str(install)
+        if index is not None:
+            url = "/".join((self._make_url(), str(index), "resolve"))
+        else:
+            url = "/".join((self._make_url(), "resolve"))
+        return self._get(url=url, params=params)
+
+    def resolve_toolbox(
+        self,
+        index: Optional[int] = None,
+        tool_ids: Optional[List[str]] = None,
+        resolver_type: Optional[str] = None,
+        container_type: Optional[str] = None,
+        requirements_only: bool = False,
+        install: bool = False,
+    ) -> list:
+        """
+        Apply resolve() to each tool in the toolbox and return the results as a list.
+        See documentation for resolve() for a description of parameters that can be
+        consumed and a description of the resulting items.
+
+        :type index: int
+        :param index: index of the dependency resolver with respect to
+            the dependency resolvers config file
+
+        :type tool_ids: list
+        :param tool_ids: tool_ids to filter toolbox on
+
+        :type resolver_type: str
+        :param resolver_type: restrict to specified resolver type
+
+        :type container_type: str
+        :param container_type: restrict to specified container type
+
+        :type requirements_only: bool
+        :param requirements_only: ignore tool containers, properties - just search based on tool requirements set to True to mimic default behavior of tool dependency API.
+
+        :type install: bool
+        :param install: allow installation of new containers (for build_mulled containers) the way job resolution will operate, defaults to False
+
+        :rtype: list
+          For example::
+        [{'tool_id': 'upload1', 'status': {'model_class': 'NullDependency', 'dependency_type': None, 'exact': True, 'name': None, 'version': None, 'cacheable': False}, 'requirements': []}, ...]
+        """
+        params = {}
+        if tool_ids:
+            params["tool_ids"] = ",".join(tool_ids)
```

Comment:
You can just pass in a list, but I'd send it as json.

Reply:
Is there somewhere an example, of how to do this? That would be cool .. in my tests I already had problems with the length.

Reply:
Hmm, there's some wrapping going on here around the requests library. You'd normally set `json=True` and there's the caveat of https://stackoverflow.com/questions/978061/http-get-with-request-body. Given the things this call does maybe a POST would have been more appropriate.